File: skill_sdk/tests/fixtures/java-project/src/main/java/com/fixture/shop/payment/MockGateway.java
package com.fixture.shop.payment;

/**
 * Always-succeeds mock gateway for testing.
 */
public class MockGateway implements PaymentGateway {

    private int callCount = 0;

    @Override
    public PaymentResult charge(double amount, String currency) {
        callCount++;
        return new PaymentResult(true, "mock_tx_" + callCount,
                "Mock charge " + amount + " " + currency);
    }

    @Override
    public PaymentResult refund(String transactionId) {
        return new PaymentResult(true, "mock_ref_" + transactionId,
                "Mock refund " + transactionId);
    }

    public int getCallCount() {
        return callCount;
    }
}

File: skill_sdk/tests/fixtures/ts-project/src/shop/payment/MockGateway.ts
import { PaymentGateway } from './PaymentGateway';
import { PaymentResult } from './PaymentResult';

/**
 * Always-succeeds mock gateway for testing.
 */
export class MockGateway implements PaymentGateway {
  private callCount = 0;

  charge(amount: number, currency: string): PaymentResult {
    this.callCount++;
    return new PaymentResult(true, `mock_tx_${this.callCount}`, `Mock charge ${amount} ${currency}`);
  }

  refund(transactionId: string): PaymentResult {
    return new PaymentResult(true, `mock_ref_${transactionId}`, `Mock refund ${transactionId}`);
  }

  getCallCount(): number {
    return this.callCount;
  }
}
